botocore/utils.py

```diff
@@ -364,6 +364,23 @@ def parse_to_aware_datetime(value):
     return datetime_obj
 
 
+def datetime2timestamp(dt):
+    """Calculate the timestamp based on the given datetime instance.
+
+    Naive datetime input will be treated as UTC time rather than local time.
+    Note that this behavior is different from datetime.timestamp() in Python 3.
+    (We want unambiguous UTC to avoid test cases failing on different locale.)
+    """
+    epoch = datetime.datetime(1970, 1, 1)
+    if dt.tzinfo:
+        d = dt.replace(tzinfo=None) - dt.utcoffset() - epoch
```

Comment:
Correct me if I am wrong, I thought there is no UTC TZ defined in builtin datetime in Python. So I always tend to write it in a way to avoid the dependency to "from dateutil.tz import tzutc".

Besides, the current logic is kinda documented in the standard python document too. Just search ``converted to naive UTC datetimes first. The result is (a.replace(tzinfo=None) - a.utcoffset())`` in [this page](https://docs.python.org/2/library/datetime.html).